### PR TITLE
Feature/89 support stored proc

### PR DESCRIPTION
**Pull request by souravroy**
Support for Stored procedure and function calls.

### CODE REVIEW

**Comment by thadguidry** [comment]
In the future, you can use GitHub web interface to set a PR as a Draft, instead of regular PR.  I've done this for you now however.

**Comment by souravroy** [comment]
Thank you for letting me know @thadguidry

File: pom.xml
Comment (kdhrubo) [inline pom.xml:1]: Do we need this dependency. The starter is already present

<dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-jdbc</artifactId>
        </dependency>

File: src/main/java/com/homihq/db2rest/rest/rpc/SubRoutine.java
Comment (kdhrubo) [inline src/main/java/com/homihq/db2rest/rest/rpc/SubRoutine.java:1]: What is the advantage of validating the subroutine? Does this require an additional trip to the database?
Is it not worthwhile to just package the variables and let database verify everything as it is the key function of db.

File: src/main/java/com/homihq/db2rest/rest/rpc/SubRoutine.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    Map<String, Object> callSubRoutine(String subRoutineName, Map<String, Object> inParams) {
+        jdbcTemplate.setResultsMapCaseInsensitive(true);
+        SqlParameterSource in = new MapSqlParameterSource()
+                .addValues(inParams);
+        return getSimpleJdbcCall(subRoutineName).execute(in);
+    }
+
+    boolean isInvalidSubroutine(String subRoutineName) throws SQLException {
+        try(var connection = Objects.requireNonNull(jdbcTemplate.getDataSource()).getConnection()) {
```
Comment (kdhrubo) [inline src/main/java/com/homihq/db2rest/rest/rpc/SubRoutine.java:34]: What happens to the connection and resultset object? are they closed automatically?

**Review by kdhrubo — CHANGES_REQUESTED** [review]
Looks good in most parts. Please review and the comments and we should merge this today.

File: src/main/java/com/homihq/db2rest/rest/rpc/SubRoutine.java
Comment (kdhrubo) [reply src/main/java/com/homihq/db2rest/rest/rpc/SubRoutine.java:1]: Lets remove the call and rely on the db. 
One of the goal of db2rest is to be blazing fast.

File: pom.xml
Comment (souravroy) [reply pom.xml:1]: > Do we need this dependency. The starter is already present
> 
> org.springframework.boot spring-boot-starter-jdbc

Removed `spring-jdbc` dependency. Earlier it won't compile without this jar, I'm not sure what has changed, it's perfectly fine without `spring-jdbc` this time

File: src/main/java/com/homihq/db2rest/rest/rpc/SubRoutine.java
Comment (souravroy) [reply src/main/java/com/homihq/db2rest/rest/rpc/SubRoutine.java:1]: Removed validation

**Review by kdhrubo — APPROVED** [review]

**Comment by thadguidry** [comment]
@souravroy @kdhrubo It's often better to just use try-with-resources.  As long as you declare and init inside the `try` then those resources will be auto-closed after the `try` execution.  You can even have effectively final variables as of Java 9.
https://www.baeldung.com/java-try-with-resources

**Pull request merged**